plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerResolverJob.java

```diff
@@ -40,6 +44,11 @@ public LibraryClasspathContainerResolverJob(IJavaProject javaProject) {
 
   @Override
   protected IStatus run(IProgressMonitor monitor) {
+    // may have been deleted before this job runs
+    if (!javaProject.exists()) {
```

Comment:
This addition looks important in itself. Is that so?

Reply:
It was to remedy some NPEs occurring during test tear-down.  BuildPath starts a library resolver job (this job) to generate the serialized container cache file, but the test may be finished and the project deleted by the time this job starts to execute.